Userland/Services/SystemServer/Service.cpp

```diff
@@ -198,13 +198,19 @@ ErrorOr<void> Service::spawn(int socket_fd)
             TRY(Core::System::setenv("SOCKET_TAKEOVER"sv, socket_takeover_builder.string_view(), true));
         }
 
-        if (m_account.has_value() && m_account.value().uid() != getuid()) {
-            auto& account = m_account.value();
-            if (auto error_or_void = account.login(); error_or_void.is_error()) {
-                dbgln("Failed to drop privileges (GID={}, UID={}), due to {}\n", account.gid(), account.uid(), error_or_void.error());
-                exit(1);
+        // NOTE: Dropping privileges makes sense when SystemServer is running
+        // for a root session.
+        // This could happen when we need to spawn a Service to serve a
+        // client with non-user UID/GID.
+        if (auto current_uid = getuid(); current_uid == 0) {
+            if (m_account.has_value() && m_account.value().uid() != current_uid) {
```

Comment:
Maybe these two `if`s should be switched... in case where `m_account` retrieves a value from the `User=...` INI setting, we would have a configured intent to run the service as a certain user. If we're not able to fulfill that request, we should fail imho.

On the other hand, in your case the value for `m_account` is determined by the socket owner I think? And in that case, it would be fine to ignore a failure since we're assuming we need to run the service as a different user when in fact we can't do that as a non-root user (and we certainly can't drop to `root`).